### DISCUSSION TITLE
Teleport service hanging

### DISCUSSION THREAD

**Discussion by stuartblackburn** [gravitational/teleport#17944] [Q&A] [answered]
Hi all,

My company supports a client who uses teleport for system access. Personally, I have very limited knowledge of the product, but I'm trying to figure out why the service stops allowing logins on servers periodically. I've enabled diagnostic mode on all of the servers in an attempt to troubleshoot, but when the service is in this problematic state, the service is still up and the diagnostic check URLs (http://127.0.0.1:3000/readyz, http://127.0.0.1:3000/healthz) both return ok status code 200 results.

I've generated a debug dumb by following the instructions listed in https://goteleport.com/docs/management/admin/troubleshooting/. My service is still in INFO mode though, not DEBUG because I don't want to restart it until I figure out the cause of the problem. 

I'm not seeing anything in this dump that is an obvious sign of the problem though. Is there something specific someone might be able to suggest I look for? Due to security reasons I'm not able to copy the dump outside of the system.

Here is my version info:

v4.3.2 git:v4.3.2-0-gc6d702ad8 go1.13.2

Thanks!

**Comment by webvictim** [comment, marked as answer]
Teleport 4.3.2 is very old now and is likely to have a number of bugs which have been fixed. We'd obviously recommend upgrading Teleport to a newer version (we're up to v11 now) if possible.

In the short term, if you could share either the dump or the logs from the process at `INFO` level we can probably figure out what's happening. I understand that security restrictions suggest that information shouldn't be copied outside the system, but there's nothing in the debug dump which could identify the system or provide an attack vector.

If it's really not at all possible to share the logs, we can't really help. You can take a look at how long goroutines have existed (each should have a count in minutes at the start) and see whether any appear to be blocked.

**Reply by stuartblackburn** [reply]
Hi, thanks for the reply! I checked with the client about upgrading the teleport version and it's currently being looked in to. Also managed to grab debug dump off the server. Here it is attached for reference. 
[syslog.txt](https://github.com/gravitational/teleport/files/9903197/syslog.txt)

**Reply by webvictim** [reply]
Not too much I can pull out of that unfortunately. It appears to be blocked on waiting for something, but it's pretty unclear what that could be.

Unfortunately on such an old version there's very little guidance I can provide. The best solution for now is probably to put in a cron job to restart the Teleport process once a day during a time when access won't be taking place.

**Reply by stuartblackburn** [reply]
Ok I'll inquire with the client about potentially upgrading everything to a newer version. Thanks very much for investigating this!
